### PR TITLE
Create helper for fulcio config yaml

### PR DESCRIPTION
**Pull request by javanlacerda**
## Description of the change

I noticed that [this](https://github.com/sigstore/helm-charts/blob/b0a2dc97ee241bf1ab8b01726a48261108610df0/charts/fulcio/templates/fulcio-configmap.yaml#L16) condition was not working properly when I was bumping the fulcio version in scaffold.

Then I managed to move the logic of create the config with yaml format into the helper and setting the indent correctly.

## Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). Where applicable, update and bump the versions in any associated umbrella chart
- [x] Variables are documented in the `values.yaml` and added to the README.md. The [helm-docs](https://github.com/norwoodj/helm-docs) utility can be used to generate the necessary content. Use `helm-docs --dry-run` to preview the content.
- [x] JSON Schema generated.
- [x] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command.

### CODE REVIEW

File: charts/fulcio/templates/fulcio-configmap.yaml [hunk 1]
```diff
@@ -10,12 +10,10 @@ data:
 # config's content.
 # If the field format is empty, the default case is consider that is a
 # json or is empty and should use the defaults as defined on the file _helpers.tpl
-{{- if and (eq .Values.config.format "yaml") (.Values.config.contents) -}}
-  {{- with .Values.config.contents }}
+{{- if and (eq .Values.config.format "yaml")}}
```
Comment (haydentherapper) [inline charts/fulcio/templates/fulcio-configmap.yaml:13]: Do you need “and”?

**Review by haydentherapper — COMMENTED** [review]
Just wondering, what was the error you were getting with the previous code?

File: charts/fulcio/templates/fulcio-configmap.yaml (hunk 1 above)
Comment (javanlacerda) [reply charts/fulcio/templates/fulcio-configmap.yaml:13]: No! removing

**Comment by javanlacerda** [comment]
> Just wondering, what was the error you were getting with the previous code?

I've got this one:

```Error: INSTALLATION FAILED: 1 error occurred:
        * ConfigMap in version "v1" cannot be handled as a ConfigMap: json: cannot unmarshal object into Go struct field ConfigMap.data of type string```

**Review by haydentherapper — APPROVED** [review]
To confirm, this was tested locally with json and yaml configs and was working?

**Comment by javanlacerda** [comment]
> To confirm, this was tested locally with json and yaml configs and was working?

I tested for the default value, that is json. I'm doing a double check here.

**Comment by javanlacerda** [comment]
> > To confirm, this was tested locally with json and yaml configs and was working?
> 
> I tested for the default value, that is json. I'm doing a double check here.

Tested! We're good

**Comment by javanlacerda** [comment]
> > To confirm, this was tested locally with json and yaml configs and was working?
> 
> I tested for the default value, that is json. I'm doing a double check here.

It is 100% tested now, working properly for json and yaml

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**